src/extensions/agent_modules/cloud_communication/adps2_mqtt_client/src/adps2_mqtt_client_module.c

```diff
@@ -0,0 +1,626 @@
+}
+
+void UnsubScribeToDSPModuleCommands()
+{
+    // TODO (nox-msft): unsubscribe to DPS module commands.
+    // InterModuleCommand_Unsubscribe("module.microsoft.dpsclient.ipcCommands", OnCommandReceived, NULL);
+}
+
+bool IsConnectionResetRequested()
+{
+    // TODO (nox-msft): listen for a signal to reset the connection.
+    return false;
+}
+
+/**
+ * @brief Reset the connection.
+ */
+void ResetConnection()
+{
+    // TODO (nox-msft): ensure that we tear down the connection and re-establish it.
+    s_moduleState.connected = s_moduleState.subscribed = false;
+    s_moduleState.registerState = ADPS_REGISTER_STATE_UNKNOWN;
+}
+
+/**
+ * @brief Perform the work for the extension. This must be a non-blocking operation.
+ *
+ * @return ADUC_Result
+ */
+int ADPS_MQTT_Client_Module_DoWork(ADUC_AGENT_MODULE_HANDLE handle)
+{
+    ADUC_CommunicationChannel_DoWork();
+
+    if (ADUC_CommunicationChannel_IsConnected())
+    {
+        DeviceRegistration_DoWork();
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Get the Data object for the specified key.
+ *
+ * @param handle agent module handle
+ * @param dataType data type
+ * @param key data key/name
+ * @param data return data
+ * @param size return size of the return value
+ * @return int 0 on success
+*/
+int ADPS_MQTT_Client_Module_GetData(
+    ADUC_AGENT_MODULE_HANDLE handle, ADUC_MODULE_DATA_TYPE dataType, int key, void* data, int* size)
+{
+    IGNORED_PARAMETER(handle);
+    IGNORED_PARAMETER(dataType);
+    IGNORED_PARAMETER(key);
+    IGNORED_PARAMETER(data);
+    IGNORED_PARAMETER(size);
+    int ret = 0;
+    return ret;
+}
+
+/**
+ * @brief Set the Data object for the specified key.
+ *
+ * @param handle agent module handle
+ * @param dataType data type
+ * @param key data key/name
+ * @param data data buffer
+ * @package size size of the data buffer
+ */
+int ADPS_MQTT_Client_Module_SetData(
+    ADUC_AGENT_MODULE_HANDLE handle, ADUC_MODULE_DATA_TYPE dataType, int key, void* data, int size)
+{
+    IGNORED_PARAMETER(handle);
+    IGNORED_PARAMETER(dataType);
+    IGNORED_PARAMETER(key);
+    IGNORED_PARAMETER(data);
+    IGNORED_PARAMETER(size);
+    int ret = 0;
+    return ret;
+}
+
+ADUC_AGENT_MODULE_INTERFACE ADPS_MQTT_Client_ModuleInterface = {
+    &s_moduleState,
+    ADPS_MQTT_Client_Module_Create,
+    ADPS_MQTT_Client_Module_Destroy,
+    ADPS_MQTT_Client_Module_GetContractInfo,
+    ADPS_MQTT_Client_Module_DoWork,
+    ADPS_MQTT_Client_Module_Initialize,
+    ADPS_MQTT_Client_Module_Deinitialize,
+    ADPS_MQTT_Client_Module_GetData,
+    ADPS_MQTT_Client_Module_SetData,
+};
+
+bool ADPS_MQTT_CLIENT_MODULE_IsDeviceRegistered(ADUC_AGENT_MODULE_HANDLE handle)
+{
+    return (s_moduleState.registerState == ADPS_REGISTER_STATE_REGISTERED);
+}
+
+const char* ADPS_MQTT_CLIENT_MODULE_GetMQTTBrokerEndpoint(ADUC_AGENT_MODULE_HANDLE handle)
+{
+    return json_object_dotget_string(
+        json_value_get_object(s_moduleState.registrationData), "assignedEnpoint.hostName");
```

Comment:
typo, missing a 'd': assignedEnpoint

Is DPS v2 service also missing the 'd' in their schema and needs correcting? Otherwise, how is it working as it should be failign to get the endpoint?

Reply:
ok, seems to be fixed in the latest iteration(s).